src/test/mochitest/head.js

```diff
@@ -343,13 +344,22 @@ function isPaused(dbg) {
  * @static
  */
 async function waitForPaused(dbg) {
-  // We want to make sure that we get both a real paused event and
-  // that the state is fully populated. The client may do some more
-  // work (call other client methods) before populating the state.
-  let loading = waitForDispatch(dbg, "LOAD_OBJECT_PROPERTIES");
-  await waitForThreadEvents(dbg, "paused");
-  await waitForState(dbg, state => isPaused(dbg));
-  await loading;
+  function scopeIsLoaded(dbg) {
```

Comment:
Rather than digging around in the state and counting properties, would it be possible for, say, something in `src/reducers` to provide a predicate on a state that indicates whether the scopes are fully loaded? It seems to me that duplicating the knowledge of that structure's details here is asking for trouble.